pkg/resource/db_cluster/custom_update.go

```diff
@@ -561,7 +562,12 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
 		res.SetEnableIAMDatabaseAuthentication(*desired.ko.Spec.EnableIAMDatabaseAuthentication)
 	}
 	if desired.ko.Spec.EngineVersion != nil && delta.DifferentAt("Spec.EngineVersion") {
-		res.SetEngineVersion(*desired.ko.Spec.EngineVersion)
+		r := regexp.MustCompile(`[0-9]*$`)
```

Comment:
Can we delcare this regex as a total level file variable? just avoid compiling the regex at every reconciliation loop